### ISSUE TITLE
Example with pytorch lightning

### ISSUE THREAD

**Issue by shaileshj2803** [luyug/GradCache#8]
Do you have example of this with pytorch lightning by any chance?
Thanks for the beautiful work.

**Comment by luyug** [comment]
I'm glad you like this work!

Unfortunately, I have not worked with pytorch lightning before. I can probably offer some suggestions if you run into some more concrete problems.

**Comment by aaprasad** [comment]
hi @shaileshj2803 did you end up integrating this with lightning? Would love to see an example as well!

**Comment by sulekz** [comment]
Lightning offers a built-in gradient accumulation method: https://lightning.ai/docs/pytorch/latest/advanced/training_tricks.html#accumulate-gradients

Not sure if it is exactly equivalent to GradCache though but might be an easy to try alternative.

**Comment by yang-su2000** [comment]
hi @aaprasad, you can look into my recent PR for reference.
